src/senaite/core/upgrade/v02_04_000.py

```diff
@@ -71,3 +74,37 @@ def reindex_qc_analyses(tool):
         obj._p_deactivate()
 
     logger.info("Reindexing QC Analyses [DONE]")
+
+
+def mark_retracted_and_rejected_analyses(portal):
```

Comment:
Note: When calling from a setup step, the parameter here is the `portal_setup` tool, not the portal object

Reply:
Thanks!. Fixed with https://github.com/senaite/senaite.core/pull/2164/commits/8fa68c346bd9d08f24a1535d1b7e966e7932201d